public/app/core/components/TraceToProfiles/TraceToProfilesSettings.tsx

```diff
@@ -0,0 +1,187 @@
+
+import { TagMappingInput } from '../TraceToLogs/TagMappingInput';
+export interface TraceToProfilesOptions {
+  datasourceUid?: string;
+  tags?: Array<{ key: string; value?: string }>;
+  query?: string;
+  profileTypeId?: string;
+  customQuery: boolean;
+}
+
+export interface TraceToProfilesData extends DataSourceJsonData {
+  tracesToProfiles?: TraceToProfilesOptions;
+}
+
+interface Props extends DataSourcePluginOptionsEditorProps<TraceToProfilesData> {}
+
+export function TraceToProfilesSettings({ options, onOptionsChange }: Props) {
+  const supportedDataSourceTypes = useMemo(() => ['grafana-pyroscope-datasource'], []);
+
+  const [profileTypes, setProfileTypes] = useState<ProfileTypeMessage[]>([]);
+  const profileTypesPlaceholder = useMemo(() => {
+    let placeholder = profileTypes.length === 0 ? 'No profile types found' : 'Select profile type';
+    if (!options.jsonData.tracesToProfiles?.datasourceUid) {
+      placeholder = 'Please select profiling data source';
+    }
+    return placeholder;
+  }, [options.jsonData.tracesToProfiles?.datasourceUid, profileTypes]);
+
+  const { value: dataSource } = useAsync(async () => {
+    return await getDataSourceSrv().get(options.jsonData.tracesToProfiles?.datasourceUid);
+  }, [options.jsonData.tracesToProfiles?.datasourceUid]);
+
+  useEffect(() => {
+    if (
+      dataSource &&
+      dataSource instanceof PyroscopeDataSource &&
+      supportedDataSourceTypes.includes(dataSource.type) &&
+      dataSource.uid === options.jsonData.tracesToProfiles?.datasourceUid
+    ) {
+      dataSource.getProfileTypes().then((profileTypes) => {
+        setProfileTypes(profileTypes);
+      });
+    } else {
+      setProfileTypes([]);
+    }
+  }, [dataSource, onOptionsChange, options, supportedDataSourceTypes]);
+
+  return (
+    <div className={css({ width: '100%' })}>
+      <InlineFieldRow>
+        <InlineField
+          tooltip="The profiles data source the trace is going to navigate to"
+          label="Data source"
+          labelWidth={26}
+        >
+          <DataSourcePicker
+            inputId="trace-to-profiles-data-source-picker"
+            filter={(ds) => supportedDataSourceTypes.includes(ds.type)}
+            current={options.jsonData.tracesToProfiles?.datasourceUid}
+            noDefault={true}
+            width={40}
+            onChange={(ds: DataSourceInstanceSettings) => {
+              console.log(options.jsonData.tracesToProfiles, ds);
+              updateDatasourcePluginJsonDataOption({ onOptionsChange, options }, 'tracesToProfiles', {
+                ...options.jsonData.tracesToProfiles,
+                datasourceUid: ds.uid,
+              });
+            }}
+          />
+        </InlineField>
+      </InlineFieldRow>
+
+      <InlineFieldRow>
+        <InlineField
+          tooltip="Tags that will be used in the query. Default tags: 'cluster', 'hostname', 'namespace', 'pod', 'service.name', 'service.namespace', 'deployment.environment'"
```

Comment:
Are these defaults reasonable for the profiles? Seems like a lot of tags that may not exist in Pyroscope.

Reply:
Was not sure and felt no harm in leaving them in. Having said that, I've reduced them now to which we know are reasonable for profiles.

Reply:
I think if we use some that do not map to profile labels they will create a query that doesn't return anything isn't it like that? That is why I would rather not have ones that we are not sure will exist in the profiling

Reply:
Yeah that's right I believe. This has been updated.